src/main/java/Parser.java

```diff
@@ -0,0 +1,90 @@
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Parser class deals with making sense of user input.
+ */
+public class Parser {
+    
+    /**
+     * Takes in user input and processes it, producing an output.
+     * @param ui the Ui currently in use.
+     * @param taskList the TaskList currently in use.
+     * @param storage the Storage object currently in use.
+     */
+    public static void echo(Ui ui, TaskList taskList, Storage storage) {
```

Comment:
Perhaps calling it something else other than echo would be more apt? Echo suggests that Duke only echos! Whilst in the method you're doing alot more than just echoing :-)